server/src/test/java/org/elasticsearch/rest/RestControllerTests.java

```diff
@@ -634,20 +637,12 @@ public void testDispatchCompatibleHandler() {
         FakeRestRequest fakeRestRequest = requestWithContent(mediaType);
         AssertingChannel channel = new AssertingChannel(fakeRestRequest, true, RestStatus.OK);
         // dispatch to a compatible handler
-        restController.registerHandler(RestRequest.Method.GET, "/foo", new RestHandler() {
-            @Override
-            public void handleRequest(RestRequest request, RestChannel channel, NodeClient client) throws Exception {
-                // in real use case we will use exact version RestApiCompatibleVersion.V_7
-                XContentBuilder xContentBuilder = channel.newBuilder();
-                assertThat(xContentBuilder.getRestApiCompatibilityVersion(), equalTo(RestApiCompatibleVersion.minimumSupported()));
-                assertThat(request.contentParser().getRestApiCompatibleVersion(), equalTo(RestApiCompatibleVersion.minimumSupported()));
-                channel.sendResponse(new BytesRestResponse(RestStatus.OK, BytesRestResponse.TEXT_CONTENT_TYPE, BytesArray.EMPTY));
-            }
-
-            @Override
-            public RestApiCompatibleVersion compatibleWithVersion() {
-                return RestApiCompatibleVersion.minimumSupported();
-            }
+        restController.registerHandler(new Route(GET, "/foo"), (request, channel1, client) -> {
```

Comment:
so this does not indicate that a Route is compatible only. It passes because if you are using compatible api it allows you to access Current Version apis (not compatible) - this is covered in testDispatchCompatibleRequestToNewlyAddedHandler
We could possibly make that test stricker, by sending another fakeRequest for version = current and expecting 404

Reply:
Good catch -- does 65379b80fa22e8af1dc9b54c4fbe7490231addfc cover what you were hoping to see?